routine/src/clj/parts/routine/core.clj

```diff
@@ -0,0 +1,136 @@
+(ns parts.routine.core
+  (:require [parts.routine.schedulers :as schedulers]))
+
+(defn add-routines
+  [w]
+  (assoc w
+         :routines
+         (filter
+           (fn [routine]
+             (when (:routine/kind routine)
+               (assert (:routine/fn routine))
+               (assert (:routine/interval-ms routine))
+               (assert (:routine/category routine))))
+           ((:system/get-register w)))))
```

Comment:
Instead of baking `:routines` into the system map, it should be a function `(defn get-routines [w] ...)`, invokes `((:system/get-register w))` and filter for all `:routine/kind` entries